ports/nrf/common-hal/busio/I2C.c

```diff
@@ -94,11 +94,19 @@ static uint8_t twi_error_to_mp(const nrfx_err_t err) {
     return 0;
 }
 
-void common_hal_busio_i2c_construct(busio_i2c_obj_t *self, const mcu_pin_obj_t *scl, const mcu_pin_obj_t *sda, uint32_t frequency, uint32_t timeout) {
+void common_hal_busio_i2c_construct(busio_i2c_obj_t *self, const mcu_pin_obj_t *scl, const mcu_pin_obj_t *sda, uint8_t setpullup, uint32_t frequency, uint32_t timeout) {
     if (scl->number == sda->number) {
         mp_raise_ValueError(translate("Invalid pins"));
     }
 
+    // Set pulls on I2C pins up if setpullup is 1 - this supports boards with internal pullups
+    if (setpullup){
+        nrf_gpio_pin_pull_t hal_pull = NRF_GPIO_PIN_PULLUP;
+
+        nrf_gpio_cfg_input(scl->number, hal_pull);
+        nrf_gpio_cfg_input(sda->number, hal_pull);
+    }
+
```

Comment:
I think following should go in the `else` routine for this condition otherwise the pullups will reset down the line:
https://github.com/adafruit/circuitpython/blob/87cedbb66849b92896de0525fc8c8214a00293b2/ports/nrf/common-hal/busio/I2C.c#L124-L142

Reply:
`set_pullup` to `internal_pullup`, and I agree with @microDev1. You want to skip the test for existing pullups.

Reply:
Can do.

It take you to mean the suggested `else` should go after `if` not `#if`, but a little confused on this point.

I think the `#if` will always succeed on my board, which I'm gathering isn't desired.

Reply:
I'm now thinking it should be called `internal_pullups`, plural, since there is more than one. I changed my suggestions to reflect that.

Reply:
Can do. If I take your meaning correctly, you're suggesting something like:
```
// Set pulls up if setpullup is 1
    if (internal_pullup){
        nrf_gpio_pin_pull_t hal_pull = NRF_GPIO_PIN_PULLUP;

        nrf_gpio_cfg_input(scl->number, hal_pull);
        nrf_gpio_cfg_input(sda->number, hal_pull);
    }
    else if (!nrf_gpio_pin_read(sda->number) || !nrf_gpio_pin_read(scl->number)) {
        reset_pin_number(sda->number);
        reset_pin_number(scl->number);
        mp_raise_RuntimeError(translate("No pull up found on SDA or SCL; check your wiring"));
    }
```

Is this correct?

Reply:
```c
   else if (!nrf_gpio_pin_read(sda->number) || !nrf_gpio_pin_read(scl->number)) {
        reset_pin_number(sda->number);
        reset_pin_number(scl->number);
        mp_raise_RuntimeError(translate("No pull up found on SDA or SCL; check your wiring"));
    }
```
Not just those lines: pull the `if` and `else` further up so the else starts at line 124. You don't want to do any of the checking code if you are enabling internal pullups.